src/main/java/de/caritas/cob/agencyservice/config/SecurityConfig.java

```diff
@@ -64,10 +69,16 @@ public SecurityConfig(KeycloakClientRequestFactory keycloakClientRequestFactory)
   @Override
   protected void configure(HttpSecurity http) throws Exception {
     super.configure(http);
-    http.csrf().disable()
+    HttpSecurity httpSecurity = http.csrf().disable()
         .addFilterBefore(new StatelessCsrfFilter(csrfCookieProperty, csrfHeaderProperty),
-            CsrfFilter.class)
-        .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
+            CsrfFilter.class);
+
+    if (multitenancy) {
```

Comment:
please rename to multitenancyEnabled and extract block to a separate method like enableTenantFilterIfMultitenancyEnabled()

Reply:
done